candidategenerator.js

```diff
@@ -167,12 +167,42 @@ window.GEN = function(sTree, words, options){
 	return candidates;
 }
 
-function rootIsHeaded(pRoot, recCat) {
-	var children = pRoot.children || [];
-	for (var i = 0; i < children.length; i++)
-		if (children[i].cat === pCat.nextLower(pRoot.cat))
+/* Function to check if a tree obeys headedness. Each node must either be be
+ * terminal or have at least one child of the category immidately below its own
+ * on the prosodic hierarch. Otherwise, return false. Written as a recursive
+ * function, basically a constraint.
```

Comment:
Okay, this is nice implementation of a check for headedness at all levels. That's not quite what I had in mind (see issue description), but after discussing the expected use cases with Richard, I've concluded that this fix is fine and we don't need to worry about people wanting to enforce headedness at only some levels.